error[E0283]: type annotations needed

I'm getting this error when trying to build kifmm on Ubuntu with Rust 1.80.1. Full error:

```
error[E0283]: type annotations needed
    --> kifmm/src/tree/morton.rs:1363:55
     |
1363 |             assert!(ancestor.level() == current_level.try_into().unwrap());
     |                                      --               ^^^^^^^^
     |                                      |
     |                                      type must be known at this point
     |
     = note: multiple `impl`s satisfying `u64: PartialEq<_>` found in the following crates: `core`, `serde_json`:
             - impl PartialEq<serde_json::value::Value> for u64;
             - impl<host> PartialEq for u64
               where the constant `host` has type `bool`;
help: try using a fully qualified path to specify the expected types
     |
1363 |             assert!(ancestor.level() == <usize as TryInto<T>>::try_into(current_level).unwrap());
     |                                         ++++++++++++++++++++++++++++++++             ~

For more information about this error, try `rustc --explain E0283`.

```

--- a/kifmm/Cargo.toml
+++ b/kifmm/Cargo.toml
@@ -23,7 +23,6 @@ crate-type = ["cdylib", "lib"]
 [dependencies]
 itertools = "0.13.*"
 num = "0.4"
-approx = "0.5"
 rayon = "1.9"
 rand = "0.8.5"
 kifmm-fftw-sys = { path = "../fftw-sys" }
@@ -36,14 +35,11 @@ superslice = {version = "1.0.0", optional = true}
 pulp = "0.18.12"
 bytemuck = "1.16.0"
 rand_distr = "0.4.3"
-bitflags = "2.6.0"
 rlst = {version = "0.2.0"}
-coe-rs = "0.1.2"
 
 [dev-dependencies]
+approx = "0.5"
 criterion = { version = "0.5.*", features = ["html_reports"]}
-csv = "1.3.0"
-num_cpus = "1.16.0"
 
 [build-dependencies]
 cbindgen = "0.26.0"

--- a/kifmm/Cargo.toml
+++ b/kifmm/Cargo.toml
@@ -31,9 +31,9 @@ lazy_static = "1.4.0"
 num-complex = "0.4.5"
 mpi = { version = "0.8.*", optional = true}
 memoffset = {version = "0.9.1", optional = true}
-superslice = {version = "1.0.0", optional = true}
 pulp = "0.18.12"
 bytemuck = "1.16.0"
+superslice = {version = "1.0.0", optional = true}
 rand_distr = "0.4.3"
 rlst = {version = "0.2.0"}
 

--- a/kifmm/src/fmm/field_translation/source_to_target/hadamard.rs
+++ b/kifmm/src/fmm/field_translation/source_to_target/hadamard.rs
@@ -1,5 +1,6 @@
 //! Implementations of 8x8 operation during Hadamard product in FFT based M2L operations.
 use crate::{fmm::types::Isa, traits::general::Hadamard8x8};
+use bytemuck as _;
 use rlst::{c32, c64, RlstScalar};
 
 /// The 8x8 Hadamard operation, always inlined.

--- a/kifmm/src/fmm/types.rs
+++ b/kifmm/src/fmm/types.rs
@@ -1,4 +1,5 @@
 //! Data structures for kernel independent FMM
+use pulp as _;
 use std::{collections::HashMap, sync::RwLock};
 
 use green_kernels::{traits::Kernel as KernelTrait, types::EvalType};

--- a/kifmm/src/lib.rs
+++ b/kifmm/src/lib.rs
@@ -74,7 +74,7 @@
 //!
 //! ## References
 //! \[1\] Ying, L., Biros, G., & Zorin, D. (2004). A kernel-independent adaptive fast multipole algorithm in two and three dimensions. Journal of Computational Physics, 196(2), 591-626.
-#![cfg_attr(feature = "strict", deny(warnings))]
+#![cfg_attr(feature = "strict", deny(warnings), deny(unused_crate_dependencies))]
 #![warn(missing_docs)]
 #![allow(clippy::doc_lazy_continuation)]
 #![allow(clippy::macro_metavars_in_unsafe)]
@@ -109,3 +109,8 @@ pub use traits::fmm::Fmm;
 #[cfg_attr(feature = "strict", deny(warnings))]
 #[warn(missing_docs)]
 pub mod bindings;
+
+#[cfg(test)]
+mod test {
+    use criterion as _; // Hack to show that criterion is used, as cargo test does not see benches
+}

--- a/kifmm/src/tree/morton.rs
+++ b/kifmm/src/tree/morton.rs
@@ -1360,7 +1360,7 @@ mod test {
 
         // Test that all ancestors found
         for (current_level, &ancestor) in ancestors.iter().enumerate() {
-            assert!(ancestor.level() == current_level.try_into().unwrap());
+            assert!(ancestor.level() as usize == current_level);
         }
 
         // Test that the ancestors include the key at the leaf level

